src/smtml/bitwuzla_mappings.default.ml

```diff
@@ -407,8 +407,13 @@ module Fresh_bitwuzla (B : Bitwuzla_cxx.S) : M = struct
     let of_ieee_bv eb sb bv = mk_term1_indexed2 Kind.Fp_to_fp_from_bv bv eb sb
 
     (* TODO *)
-    let to_ieee_bv _ =
-      Fmt.failwith "Bitwuzla_mappings: to_ieee_bv not implemented"
+    let f32_to_i32 =
+      let args = [| Types.float 8 24 |] in
+      let ret = Types.bitv 32 in
+      let sort = B.mk_fun_sort args ret in
+      B.mk_const sort ~symbol:"f32_to_i32"
+
+    let to_ieee_bv f = mk_term2 Kind.Apply f32_to_i32 f
```

Comment:
```suggestion
    let to_ieee_bv f =
      let f_size = fp_exp_size f + fp_sig_size f in
      if f_size = 32 then
        mk_term2 Kind.Apply f32_to_i32 f
      else if f_size = 64 then 
        mk_term2 Kind.Apply f64_to_i64 f
      else assert false (* Or raise a failure with a nice error message *)
```

Maybe you could do something similar to this to extend it to support both f32 and f64 :smile: